src/core/socialprovider.js

```diff
@@ -167,7 +175,6 @@ XMPPSocialProvider.prototype.connect = function(continuation) {
 
     if (this.client) {
       this.client.end();
-      delete this.client;
```

Comment:
Awesome explanation! Thanks :)
I think calling logout() makes sense then to make sure we are cleaning up properly (plus its a bit more readable) 
Once we test that, I think this PR is good